pkg/metricshandler/metrics_handler.go

```diff
@@ -189,8 +189,12 @@ func (m *MetricsHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	contentType := expfmt.NegotiateIncludingOpenMetrics(r.Header)
 
 	// We do not support protobuf at the moment. Fall back to FmtText if the negotiated exposition format is not FmtOpenMetrics See: https://github.com/kubernetes/kube-state-metrics/issues/2022.
-	if contentType != expfmt.FmtOpenMetrics_1_0_0 && contentType != expfmt.FmtOpenMetrics_0_0_1 {
-		contentType = expfmt.FmtText
+
+	// prometheus/common/expfmt does not allow to check for OpenMetrics version, so check up for older version separately
+	openMetrics_0_0_1, _ := expfmt.NewOpenMetricsFormat(expfmt.OpenMetricsVersion_0_0_1)
```

Comment:
This is not ideal, but FormatType only exposes TypeOpenMetrics which is fmtOpenMetrics_1_0_0